### PR TITLE
When we have C++11 use std::atomic instead of ACE_Atomic

### PR DESCRIPTION
**Pull request by jwillemsen**

### CODE REVIEW

File: dds/DCPS/DataBlockLockPool.h [hunk 1]
```diff
@@ -11,6 +11,11 @@
 #include "ace/Lock_Adapter_T.h"
 #include "ace/Thread_Mutex.h"
 #include "ace/Containers_T.h"
+#if defined (ACE_HAS_CPP11)
+# include <atomic>
+#else
+# include "ace/Atomic_Op.h"
+#endif
```
Comment (iguessthislldo) [inline dds/DCPS/DataBlockLockPool.h:18]: https://github.com/objectcomputing/OpenDDS/blob/master/docs/guidelines.md#preprocessor

File: dds/DCPS/DataReaderImpl.cpp [hunk 2]
```diff
@@ -224,7 +224,7 @@ DataReaderImpl::add_association(const RepoId& yourId,
         OPENDDS_STRING(writer_converter).c_str()));
   }
 
-  if (entity_deleted_.value()) {
+  if (entity_deleted_ == true) {
```
Comment (mitza-oci) [inline dds/DCPS/DataReaderImpl.cpp:227]:
```suggestion
  if (get_deleted()) {
```

File: dds/DCPS/RTPS/Spdp.h [hunk 3]
```diff
@@ -28,7 +28,11 @@
 #include "Sedp.h"
 #include "rtps_export.h"
 
-#include "ace/Atomic_Op.h"
+#ifdef ACE_HAS_CPP11
+# include <atomic>
+#else
+# include "ace/Atomic_Op.h"
```
Comment (iguessthislldo) [inline dds/DCPS/RTPS/Spdp.h:34]:
```suggestion
#  include <atomic>
#else
#  include <ace/Atomic_Op.h>
```

File: dds/DCPS/RTPS/Spdp.h [hunk 4]
```diff
@@ -95,7 +99,13 @@ class OpenDDS_Rtps_Export Spdp : public DCPS::LocalParticipant<Sedp>
   void signal_liveliness(DDS::LivelinessQosPolicyKind kind);
 
   // Is Spdp shutting down?
-  bool shutting_down() { return shutdown_flag_.value(); }
+  bool shutting_down() {
```
Comment (iguessthislldo) [inline dds/DCPS/RTPS/Spdp.h:102]:
```suggestion
  bool shutting_down()
  {
```

File: dds/DCPS/RcObject.h [hunk 5]
```diff
@@ -7,7 +7,11 @@
 
 
 #include "dds/Versioned_Namespace.h"
-#include "ace/Atomic_Op.h"
+#ifdef ACE_HAS_CPP11
+# include <atomic>
+#else
+# include "ace/Atomic_Op.h"
```
Comment (iguessthislldo) [inline dds/DCPS/RcObject.h:13]:
```suggestion
#  include <atomic>
#else
#  include <ace/Atomic_Op.h>
```

File: dds/DCPS/ReceivedDataElementList.h [hunk 6]
```diff
@@ -8,7 +8,11 @@
 #ifndef OPENDDS_DCPS_RECEIVEDDATAELEMENTLIST_H
 #define OPENDDS_DCPS_RECEIVEDDATAELEMENTLIST_H
 
-#include "ace/Atomic_Op_T.h"
+#ifdef ACE_HAS_CPP11
+# include <atomic>
+#else
+# include "ace/Atomic_Op_T.h"
```
Comment (iguessthislldo) [inline dds/DCPS/ReceivedDataElementList.h:14]:
```suggestion
#  include <atomic>
#else
#  include <ace/Atomic_Op_T.h>
```

File: dds/DCPS/TopicDescriptionImpl.h [hunk 7]
```diff
@@ -12,7 +12,11 @@
 #include "dds/DdsDcpsTypeSupportExtC.h"
 #include "Definitions.h"
 #include "ace/SString.h"
-#include "ace/Atomic_Op.h"
+#ifdef ACE_HAS_CPP11
+# include <atomic>
+#else
+# include "ace/Atomic_Op.h"
```
Comment (iguessthislldo) [inline dds/DCPS/TopicDescriptionImpl.h:18]:
```suggestion
#  include <atomic>
#else
#  include <ace/Atomic_Op.h>
```

File: dds/DCPS/WaitSet.h [hunk 8]
```diff
@@ -20,7 +20,11 @@
 #include "dds/DCPS/TimeTypes.h"
 
 #include "ace/Thread_Mutex.h"
-#include "ace/Atomic_Op.h"
+#ifdef ACE_HAS_CPP11
+# include <atomic>
+#else
+# include "ace/Atomic_Op.h"
```
Comment (iguessthislldo) [inline dds/DCPS/WaitSet.h:26]:
```suggestion
#  include <atomic>
#else
#  include <ace/Atomic_Op.h>
```

File: dds/DCPS/WriterInfo.h [hunk 9]
```diff
@@ -18,6 +18,11 @@
 #include "DisjointSequence.h"
 #include "transport/framework/ReceivedDataSample.h"
 #include "TimeTypes.h"
+#ifdef ACE_HAS_CPP11
+# include <atomic>
+#else
+# include "ace/Atomic_Op.h"
```
Comment (iguessthislldo) [inline dds/DCPS/WriterInfo.h:24]:
```suggestion

#ifdef ACE_HAS_CPP11
#  include <atomic>
#else
#  include <ace/Atomic_Op.h>
```

File: dds/DCPS/transport/framework/TransportQueueElement.h [hunk 10]
```diff
@@ -8,6 +8,11 @@
 #ifndef OPENDDS_DCPS_TRANSPORTQUEUEELEMENT_H
 #define OPENDDS_DCPS_TRANSPORTQUEUEELEMENT_H
 
+#ifdef ACE_HAS_CPP11
+# include <atomic>
+#else
+# include "ace/Atomic_Op.h"
+#endif
```
Comment (iguessthislldo) [inline dds/DCPS/transport/framework/TransportQueueElement.h:15]:
```suggestion
#ifdef ACE_HAS_CPP11
#  include <atomic>
#else
#  include <ace/Atomic_Op.h>
#endif
```

Also this should go after the dds includes.

File: dds/DCPS/transport/tcp/TcpDataLink.h [hunk 11]
```diff
@@ -12,6 +12,11 @@
 #include "TcpTransport.h"
 #include "dds/DCPS/transport/framework/DataLink.h"
 #include "ace/INET_Addr.h"
+#ifdef ACE_HAS_CPP11
+# include <atomic>
+#else
+# include "ace/Atomic_Op.h"
```
Comment (iguessthislldo) [inline dds/DCPS/transport/tcp/TcpDataLink.h:18]:
```suggestion
#  include <atomic>
#else
#  include <ace/Atomic_Op.h>
```

**Review by iguessthislldo — CHANGES_REQUESTED** [review]

**Review by iguessthislldo — APPROVED** [review]

**Pull request merged**